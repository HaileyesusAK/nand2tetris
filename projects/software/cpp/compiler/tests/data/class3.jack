class Circle {
	method boolean is_unit_cirlce() {
		return radius = 1;
	}
}
