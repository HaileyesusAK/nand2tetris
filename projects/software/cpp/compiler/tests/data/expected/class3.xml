<class>
    <keyword> class </keyword>
    <identifier> Circle </identifier>
    <symbol> { </symbol>
    <subroutineDec>
        <keyword> method </keyword>
        <keyword> boolean </keyword>
        <identifier> is_unit_cirlce </identifier>
        <symbol> ( </symbol>
        <symbol> ) </symbol>
        <subroutineBody>
            <symbol> { </symbol>
            <statements>
                <returnStatement>
                    <keyword> return </keyword>
                    <expression>
                        <term>
                            <identifier> radius </identifier>
                        </term>
                        <symbol> = </symbol>
                        <term>
                            <integerConstant> 1 </integerConstant>
                        </term>
                    </expression>
                    <symbol> ; </symbol>
                </returnStatement>
            </statements>
            <symbol> } </symbol>
        </subroutineBody>
    </subroutineDec>
    <symbol> } </symbol>
</class>
